communication/src/AsyncTlsConnection.cpp

```diff
@@ -471,9 +470,11 @@ const std::string AsyncTlsConnection::decryptPK(const boost::filesystem::path& p
   std::string pkpath;
 
   std::unique_ptr<ISecretsManagerImpl> secrets_manager;
-  if (tlsTcpImpl_.config_.secretData) {
+  if (config_.secretData) {
     pkpath = (path / fs::path("pk.pem.enc")).string();
-    secrets_manager.reset(new SecretsManagerEnc(tlsTcpImpl_.config_.secretData.value()));
+    if (secrets_manager == nullptr) {
```

Comment:
nit: Is this check needed? The pointer is created on line 472/473. I see no way to be initialised with something.

Reply:
Good catch!